packages/block-library/src/navigation-link/edit.js

```diff
@@ -167,7 +167,10 @@ function NavigationLinkEdit( {
 							} = {} ) => setAttributes( {
 								title: escape( newTitle ),
 								url: newURL,
-								label: label || escape( newTitle ),
+								label: (
+									( newTitle !== '' && newTitle !== newURL && label !== newTitle ) ?
```

Comment:
If I try replacing the link with a url without explicitly typing the protocol, it will replace the label. This seems to be because the protocol gets added automatically to the url, but not to the title, so url `!==` title. So say I have a nav item with the manually added label `Personal website`. If I edit the link to `https://tellthemachines.com`, the label doesn't change. BUT if I edit the link to `tellthemachines.com`, the label then changes to `tellthemachines.com`. 

Perhaps we could remove the protocol from newUrl before making the comparison with newTitle?

Reply:
wow good catch :) I'll see to it asap!

Reply:
fixed!